DS-Assignment1/Question 1&2/src/com/metacube/myarraylist/MyArrayList.java

```diff
@@ -0,0 +1,317 @@
+			added = false;
+		} else {
+			Comparable[] newList = new Comparable[list.length];
+			for (int index = 0; index < position; index++) {
+				newList[index] = list[index];
+			}
+			newList[position] = element;
+			for (int index = position + 1; index < list.length; index++) {
+				newList[index] = list[index - 1];
+			}
+			list = (T[]) newList;
+		}
+		return added;
+	}
+
+	/**
+	 * Removes the element from given position
+	 *
+	 * @param index
+	 *            the index
+	 * @return boolean - true if removed from list else false
+	 */
+	@Override
+	public boolean remove(int position) {
+		boolean removed = true;
+		if (position > size() || position < 0) {
+			removed = false;
+		} else {
+			int index;
+			for (index = position; index < list.length - 1; index++) {
+				list[index] = list[index + 1];
+			}
+			list[index] = null;
+			this.size--;
+		}
+		return removed;
+	}
+
+	/**
+	 * Removes the element
+	 *
+	 * @param element
+	 *            the element
+	 */
+	@Override
+	public boolean remove(T element) {
+
+		return remove(getFirstIndex(element));
+	}
+
+	/**
+	 * Gets the first index after given position
+	 *
+	 * @param element
+	 *            the element
+	 * @param position
+	 *            the index
+	 * @return the first index after position or -1 if element not found
+	 */
+	@Override
+	public int getFirstIndexAfterPosition(T element, int position) {
+		int index = 0;
+		if (position > this.size || element == null) {
+
+			return -1;
+		}
+		for (Object elementValue : list) {
+			if (index > this.size) {
+				index = -1;
+			}
+			if (index > position) {
+				if (elementValue.equals(element)) {
+					break;
+				}
+			}
+			index++;
+		}
+
+		return index;
+	}
+
+	/**
+	 * Gets the first index of element
+	 *
+	 * @param element
+	 *            the element
+	 * @return the first index of element or -1 if element not found
+	 */
+	@Override
+	public int getFirstIndex(T element) {
+		int index = 0;
+		boolean found = false;
+		for (Object elementValue : list) {
+			if (index > size()) {
+				break;
+			}
+			if (elementValue.equals(element)) {
+				found = true;
+				break;
+			}
+			index++;
+		}
+		if (found == true) {
+			return index;
+		}
+		return -1;
+	}
+
+	/**
+	 * Clear the list.
+	 */
+	@Override
+	public void clear() {
+		list = (T[]) new Comparable[FIXED_SIZE];
+		this.size = -1;
+	}
+
+	/**
+	 * Reverse the list
+	 */
+	@Override
+	public void reverse() {
+		for (int i = 0; i <= size() / 2; i++) {
+			swap(i, size() - i);
+		}
+	}
+
+	/**
+	 * Swap of the number present at first index and second index
+	 * 
+	 * @param firstIndex
+	 * @param secondIndex
+	 */
+	private void swap(int firstIndex, int secondIndex) {
+		T temp = list[firstIndex];
+		list[firstIndex] = list[secondIndex];
+		list[secondIndex] = temp;
+	}
+
+	/**
+	 * Sort the list according to the sorting order.
+	 *
+	 * @param sortingOrder
+	 *            the sorting order - Ascending or Descending
+	 */
+	@Override
+	@SuppressWarnings({ "unchecked" })
```

Comment:
Instead of suppressing warning, first look to know the cause of warning and try to fix that.